src/main/java/io/javalin/core/compression/GzipWrapper.kt

```diff
@@ -0,0 +1,31 @@
+package io.javalin.core.compression
+
+import java.io.OutputStream
+import java.util.zip.GZIPOutputStream
+
+
+/**
+ * Kotlin wrapper for Java's GZIPOutputStream.
+ * DynamicCompressionStrategy uses this to perform Gzip compression
+ * @see DynamicCompressionStrategy
+ *
+ * @param out the output stream
+ * @param syncFlush
+ *        if {@code true} invocation of the inherited
+ *        {@link DeflaterOutputStream#flush() flush()} method of
+ *        this instance flushes the compressor with flush mode
+ *        {@link Deflater#SYNC_FLUSH} before flushing the output
+ *        stream, otherwise only flushes the output stream
+ * @param level Compression level. Range 0..9, default = 6
+ */
+class GzipWrapper(val out: OutputStream,
```

Comment:
As I mentioned earlier, making `GzipWrapper` objects feels weird to me. It could be a utility class that has a level (which is checked when setting it), and a method to produce streams.

Reply:
see https://github.com/tipsy/javalin/pull/652#discussion_r302855842